packages/next-swc/crates/core/src/server_actions.rs

```diff
@@ -1692,6 +1550,45 @@ fn collect_idents_in_stmt(stmt: &Stmt) -> Vec<Id> {
     ids
 }
 
+pub(crate) struct ClosureActionReplacer<'a> {
+    replaced_action_proxies: &'a Vec<(Ident, Expr)>,
```

Comment:
```suggestion
    replaced_action_proxies: &'a Vec<(Ident, Box<Expr>)>,
```

in the name of the performance.

Reply:
If we don't box it, we have to pay for `memmove` call because `Expr` is a quite large type.

Reply:
We are going to store it as `Box<Expr>` anyway (in the final AST), so it's better to allocate it upfront